src/util/misc/boundingBoxFromPoints.ts

```diff
@@ -1,21 +1,30 @@
-import { Point } from '../../point.class';
+import { IPoint, Point } from '../../point.class';
 
 /**
  * Calculates bounding box (left, top, width, height) from given `points`
  * @static
  * @memberOf fabric.util
- * @param {Point[]} points
+ * @param {IPoint[]} points
  * @return {Object} Object with left, top, width, height properties
  */
-export const makeBoundingBoxFromPoints = (points: Point[]) => {
+export const makeBoundingBoxFromPoints = (points: IPoint[]) => {
+  if (points.length === 0) {
+    return {
+      left: 0,
+      top: 0,
+      width: 0,
+      height: 0,
+    };
+  }
```

Comment:
when this actually happens?
i would prefer a default value for points rather than a specific case in the code.

```suggestion
const defaultPoints = [{ x: 0, y: 0 }];
export const makeBoundingBoxFromPoints = (points: IPoint[] = defaultPoints) => {
```

can't TS help here saying an empty array isn't good?

Reply:
yeah my suggestion is garbage, because i wasn't thinking empty array doesn't trigger the default.

what about this stack overflow finding?
```
type NonEmptyArray<T> = T[] & { 0: T };
````

Reply:
Not sure we should go this way because I think (not sure) that TS will bug us in case of an array that you filtered or sliced or whatever since it won't know that surely it has a first element. You will need to cast it and that's annoying.
But this is not a TS thing only. Should cover all cases.